### ISSUE TITLE
Unable to run adb commands on UZ801

### ISSUE THREAD

**Issue by akash-roboticist** [OpenStick/OpenStick#16]
I was super excited to see this awesome hack (thank you for all the notes, they are awesome!) and found a promising stick that I purchased from alibaba; the primary chip checks out – MSM8916 and going by the markings on the hynix chip next to it, it seems to be a 512MB RAM, 4GB eMMC chip, so I was feeling lucky.

But the challenge is, I’m not able to get an adb connection to it. The device is pingable and UI is accessible on `192.168.100.1`, now here’s something new – there is no “upgrade” function in the UI either and after a bit of tinkering, found it is running a Eclipse Jetty webserver, rather than I what I can guess the others are running – an android app.

I also tried doing a port scan, but that didnt yield any other open port than port 80 (for UI) and port 53 (for DNS).

```
Starting Nmap 7.80 ( https://nmap.org ) at 2022-08-25 14:17 +08
Nmap scan report for 192.168.100.1
Host is up (0.0096s latency).
Not shown: 65533 closed ports
PORT   STATE SERVICE VERSION
53/tcp open  domain  dnsmasq 2.51
| dns-nsid: 
|_  bind.version: dnsmasq-2.51
80/tcp open  http    Jetty 8.y.z-SNAPSHOT
|_http-server-header: Jetty(8.y.z-SNAPSHOT)
|_http-title: 4G Modem
```

On connecting to USB, it lists as a RNDIS device, disconnects and reconnects again in a few seconds (this doesnt seem to be power related as I have tried it with a few different computers, and also, the reconnect is pretty fast ~0.25s - so it may be some mode switching internally i think)
```
[Aug20 15:04] usb 3-8.1.4: new high-speed USB device number 41 using xhci_hcd
[ +0.101530] usb 3-8.1.4: New USB device found, idVendor=05c6, idProduct=f00e, bcdDevice=ff.ff
[ +0.000010] usb 3-8.1.4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ +0.000003] usb 3-8.1.4: Product: Android
[ +0.000002] usb 3-8.1.4: Manufacturer: Android
[ +0.000002] usb 3-8.1.4: SerialNumber: 0123456789ABCDEF
[ +0.006808] rndis_host 3-8.1.4:1.0 usb0: register ‘rndis_host’ at usb-0000:00:14.0-8.1.4, RNDIS device, ce:12:d8:cd:30:e0
[ +8.630068] usb 3-8.1.4: USB disconnect, device number 41
[ +0.000192] rndis_host 3-8.1.4:1.0 usb0: unregister ‘rndis_host’ usb-0000:00:14.0-8.1.4, RNDIS device
[ +0.257230] usb 3-8.1.4: new high-speed USB device number 42 using xhci_hcd
[ +0.100923] usb 3-8.1.4: New USB device found, idVendor=05c6, idProduct=f00e, bcdDevice=ff.ff
[ +0.000009] usb 3-8.1.4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[ +0.000004] usb 3-8.1.4: Product: Android
[ +0.000002] usb 3-8.1.4: Manufacturer: Android
[ +0.000002] usb 3-8.1.4: SerialNumber: 0123456789ABCDEF
[ +0.004680] rndis_host 3-8.1.4:1.0 usb0: register ‘rndis_host’ at usb-0000:00:14.0-8.1.4, RNDIS device, 3e:d1:02:cf:1d:75
```
And `lsusb` to lists a single device
```shell
$ lsusb | grep 05c6
Bus 003 Device 062: ID 05c6:f00e Qualcomm, Inc.
```

I have tried `adb wait-for-any-recovery && adb reboot bootloader` to see if that helps with catching the stick during its boot up and also during the factory reset (via UI).

I’m a fairly new to adb / android rooting and perhaps I’m missing something very obvious – any insights? Perhaps this PCB, which seems to be a UZ801 clone (FY_UZ801_V2.1) has some test points which I could short to force boot into boot loader? I did try probing the test points for UART console by connecting an serial RX line to them - no luck there.

Photos for reference - 
PCB:
![2022-08-20_15:19:17_001](https://user-images.githubusercontent.com/7358106/186586986-53751048-899e-42db-889d-349de55613f3.png) 
![2022-08-20_15:15:10_001](https://user-images.githubusercontent.com/7358106/186586990-738ba9a6-202b-4a34-a210-62c07b5d6a6f.png)

Enclosure:
![2022-08-28_22:47:10_001](https://user-images.githubusercontent.com/7358106/187080086-3d916b47-890d-4ff5-a96b-c94457d9ed19.png)

![2022-08-28_22:46:46_001](https://user-images.githubusercontent.com/7358106/187080110-b4d44ea8-f7d2-4558-98e9-718d4f414a2d.png)

**Comment by colemickens** [comment]
Would you mind showing the full enclosure too? These ali{express,baba} listings are so inaccurate, it'd be nice to start having a mapping of what listings map to which boards.

**Comment by akash-roboticist** [comment]
Sure, updated above.
Although, from what I've noticed, the enclosures all look almost the same, so that likely wont be very helpful.

**Comment by xandark** [comment]
I bought a similar stick and was able to install OpenStick on it easily as long as you follow these directions exactly:

https://extrowerk.com/2022/07/31/OpenStick/

As a side note, I completely ignored the web UI of the original device and I never even logged into it. 

Running Linux, you may need to implement the first answer in this thread:

https://android.stackexchange.com/questions/58187/adb-can-discover-devices-but-not-fastboot

**Comment by akash-roboticist** [comment]
Yep tried the above (and extra steps), no dice; hence the additional syslogs for clues.
@xandark when you plugin your stick into the usb, what does it show up as in the syslog? That info and `lsusb` logs would be helpful here.

**Comment by skandalfo** [comment]
I got two other similar devices, in black. The web interface's "device info" page shows model to be UZ801-V1.8 for the one that has three screws and a TF reader port (under the sim reader, same package), and UZ801-V2.0 for the other, that doesn't have the TF port, and only has the two screws on the side of the USB port. None of them has the push button populated.

Maybe that button can be pushed on startup to enter a different boot mode?

Edit: forgot to note none of my two devices ever shows on ADB either.

**Comment by akash-roboticist** [comment]
I was able to boot to EDL by shorting D+ with GND; but I've not yet had luck reading the emmc as described in https://github.com/bkerler/edl#for-emmc-flash

**Comment by xandark** [comment]
> Yep tried the above (and extra steps), no dice; hence the additional syslogs for clues. @xandark when you plugin your stick into the usb, what does it show up as in the syslog? That info and `lsusb` logs would be helpful here.

@akash-roboticist I wish I could tell you as I accidentally bricked my stick through my fault alone and now it won't boot. So I'll be curious to see if I can recover it as you just mentioned with the shorting method.

**Comment by Mio-sha512** [comment]
I have figured out how to access ADB on the UZ801_v2.1
It's simply hosted on port 7628 so you can connect with:
`adb connect 192.168.100.1:7628`
Then you can use it as normal.

I've also gone ahead and made backups of the original eMMC and uploaded it to GitHub here: 
https://github.com/Mio-sha512/openstick-stuff/releases
And I will soon be uploading a Debian and OpenWRT image with install instructions so stay tuned.

Thanks, @akash-roboticist for the EDL hint.

**Comment by adrian-bl** [comment]
Another way to enable ADB seems to be by visiting http://192.168.100.1/usbdebug.html

This will return an empty page, but cause the webserver to run 'setprop ...' to enable ADB normally.

**Comment by ddscentral** [comment]
I have two UZ801 sticks (one with black and one with white shell, but identical otherwise) with V3.0 PCB which looks different from the previous models. Thanks for the tip @adrian-bl about that \/usbdebug.html URL. It did indeed work to enable ADB on my device.
Interestingly, the firmware on my stick is very recent, kernel build date is February 2nd, 2023. So these devices do still appear to be getting firmware updates.

**Comment by jendraljack** [comment]
Nice, .. i have version fy_uz801_v2.1.
just i want to know.
how many cores usage in kernel.
i get result only 2 cpu.
![Screenshot_20230406-024440](https://user-images.githubusercontent.com/4941910/230189007-b8dd8a94-bc59-49b6-9d68-aeeb73bf894d.png)

**Comment by ddscentral** [comment]
Interesting Mine shows all 4 cores in Debian. I'll check the second stick which still has stock ROM. It's late night here, so I can't check it now, will check tomorrow.
Maybe that's how the original ROM tries to work around heat issues on these sticks. By disabling half the cores.
I had to mod my Debian stick with some thermal pads and a small heatsink to keep the SoC from reaching 100C during intensive WiFi and LTE usage.

**Comment by ddscentral** [comment]
Indeed, that's the case. Core count is capped to 2 in kernel in stock ROM.
![disabled_cores](https://user-images.githubusercontent.com/26443023/230292912-4c0ea6d3-27d2-40d7-ab34-4419d4d3786f.png)
As I mentioned, I highly suspect this is done to reduce overheating.
You can use all cores in Debian.

**Comment by jendraljack** [comment]
> Indeed, that's the case. Core count is capped to 2 in kernel in stock ROM. ![disabled_cores](https://user-images.githubusercontent.com/26443023/230292912-4c0ea6d3-27d2-40d7-ab34-4419d4d3786f.png) As I mentioned, I highly suspect this is done to reduce overheating. You can use all cores in Debian.

![Uploading IMG_20230406_222749.png…]()


I'm not ready to install debian on my uz801 yet, due to device limitations. I found the line CONFIG_NR_CPUS=2 from /proc/config.gz.
 does this have anything to do with CPU usage?. from what I've read, config.gz is just a copy of the kernel when loaded on boot. so is the kernel source associated with uz801 for Stock ROMs Android.
thank you

**Comment by jendraljack** [comment]
![Uploading IMG_20230406_222749.png…]()

**Comment by jendraljack** [comment]
> Indeed, that's the case. Core count is capped to 2 in kernel in stock ROM. ![disabled_cores](https://user-images.githubusercontent.com/26443023/230292912-4c0ea6d3-27d2-40d7-ab34-4419d4d3786f.png) As I mentioned, I highly suspect this is done to reduce overheating. You can use all cores in Debian.

I'm not ready to install debian on my uz801 yet, due to device limitations. I found the line CONFIG_NR_CPUS=2 from /proc/config.gz.
 does this have anything to do with CPU usage?. from what I've read, config.gz is just a copy of the kernel when loaded on boot. i still searching link for the kernel source uz801/Android.
thank you
![IMG_20230406_222749](https://user-images.githubusercontent.com/4941910/230453151-0856411b-aad9-4851-a25b-c43b4afe7e55.png)

**Comment by ddscentral** [comment]
Yes, /proc/config.gz (if it exists) is a copy of the configuration used to build the currently running kernel.
Are you looking for sources of the stock ROMs kernel ? Don't waste your time. Even if you find out who the exact OEM of your device is, most Chinese manufacturers will outright ignore GPL requests.

"due to device limitations" - what limitations do you mean ?

**Comment by jendraljack** [comment]
> Yes, /proc/config.gz (if it exists) is a copy of the configuration used to build the currently running kernel. Are you looking for sources of the stock ROMs kernel ? Don't waste your time. Even if you find out who the exact OEM of your device is, most Chinese manufacturers will outright ignore GPL requests.
> 
> "due to device limitations" - what limitations do you mean ?

sorry, one more question. 
If i install debian ini uz801? 
How about lte band, is working like in stock Android?

**Comment by jendraljack** [comment]
![IMG-20230407-WA0023](https://user-images.githubusercontent.com/4941910/230637920-b5a29847-a0bd-429e-bc62-746927b650ab.jpg)

**Comment by ddscentral** [comment]
For me, LTE works fine in both the stock firmware and Debian.
You will need to dump the modem firmware and WiFi NVRAM files (/firmware and /persist partitions) before flashing as the firmware files included in debian.zip are specific to UFI001 board and will not work with UZ801. I have posted dumping instructions on Hackaday, but can re-post them here if you need them.

**Comment by tiernano** [comment]
Yea. Can you dump them here? Not sure what version I have but have 2: one stock one flashed, and the flashed one does not have working modem...

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ddscentral** [comment]
From /firmware partition of the stock ROM, copy image/modem.*, image/wcnss.\* and image/mba.mbn files to Debian's /lib/firmware. From image/modem_pr, pick the correct mcfg_sw.mbn according to your region/carrier and also copy it to Debian's /lib/firmware. 
From /persist partition of the stock ROM, copy WCNSS_qcom_wlan_nv.bin, to Debian's /lib/firmware/wlan/prima/
This is required for WiFi master (AP) mode to work correctly.

**Comment by jendraljack** [comment]
I successfully installed debian on my Uz801 via termux fastboot.
file from https://extrowerk.com/2022/07/31/OpenStick/

but I'm still confused with nmtui?
the led light doesn't turn on, is it like that?
![2015_01_01_07_09_52](https://user-images.githubusercontent.com/4941910/231451141-b9d47beb-e8f6-40ed-9fd8-bb84dcdf83c2.jpg)

**Comment by ddscentral** [comment]
LEDs are entirely software controlled via GPIO. What you assign to them is up to you. By default, blue is set to WiFi TX trigger, Red is set to heartbeat trigger and Green has no default trigger (I set it to mmc0 on mine to show disk activity).
Unfortunately, boot.img (kernel+initrd) from debian.zip has a DTB for the UFI001 which has LEDs assigned to different GPIO pins. There is a boot.img for UZ801 in OpenStick repo which should fix the LEDs but I haven't tried it.

You can also try my UZ801 kernel/initrd which is built using the latest sources from OpenStick repo (do install DEB files first). I've also built a fixed lk2nd bootloader (aboot.img) with "reset" button working correctly to enable fastboot mode. This offers a way to re-flash the device if it no longer boots.
https://zebra.ddscentral.org/pub/downloads/openstick/firmware/uz801_v30/kernel/
https://zebra.ddscentral.org/pub/downloads/openstick/firmware/uz801_v30/boot/

**Comment by jendraljack** [comment]
Ty, i will try new boot later.
but now ,I set wifi AP via nmtui.
but in Wifi Sta/client ,its stuck "Obtaining IP address...". not connected yet.
what i missed?.
ty ddscentral.

**Comment by ddscentral** [comment]
You can always set the IP manually to check if WiFi works.
Does "ip a" show WiFi interface as connected ?
If not, I would check "dmesg" output.

**Comment by jendraljack** [comment]
ip command?
ip command not found
but ifconfig is:
lo, usb0 and wlan0.
in nmtui, i set device to wlan0.
ssid xxxx
mode : access point.
wpa/wpa2.

**Comment by jendraljack** [comment]
yup. connected if add manually on wifi STA device.
Ty.
![Screenshot_20230412-235035](https://user-images.githubusercontent.com/4941910/231527959-40ca7a41-df08-490f-8c37-39edab3b38ea.png)

**Comment by ddscentral** [comment]
Sorry, forgot the "ip" wasn't installed by default in this Openstick's rootfs. Nowadays, usually it's "ifconfig" that's not installed by default in most distros as it's considered deprecated.
For clients to get IPs automatically in AP mode, you will need a DHCP server. Pre-installed dnsmasq server (which hands out IPs for usb0 interface, that is, the host PC) will work, but you'll need to tweak it's configuration to also hand out IPs for the WiFi interface.
BTW, do not use nmtui's AP function as it only supports ancient B and G modes. Install hostapd instead.

**Comment by mrisaev** [comment]
![dfgsf](https://github.com/OpenStick/OpenStick/assets/14234312/1b7c239b-6579-480b-9429-b56a72d364d2)
Red line - EDL
Yellow line - Fastboot

That works for UZ801_V3.2
